src/cunumeric/random/randutil/randutil_curand.h

```diff
@@ -0,0 +1,101 @@
+/* Copyright 2021-2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+
+// This header file imports some head-only part of curand for the HOST-side implementation of
+// generators
+
+// also allow usage of generators on host
+#ifdef LEGATE_USE_CUDA
+
+#define QUALIFIERS static __forceinline__ __device__ __host__
+#define RANDUTIL_QUALIFIERS __forceinline__ __device__ __host__
+#include <curand_kernel.h>
+
+#else
+// host generators are not compiled with nvcc
+#define QUALIFIERS static
+#define RANDUTIL_QUALIFIERS
+#pragma region avoid inclusion of cuda_runtime.h in curand.h, and mtgp32
+
+// include these before defining CUDACC_RTC to still enable host-side precalc matrix
+#define __device__
+#include <curand_precalc.h>
+#include <curand_mrg32k3a.h>
+#undef __device__
+
+#define __CUDACC_RTC__
```

Comment:
@fduguet-nv Did you mean to `#undef` this after line 43? I've spent quite a lot of time trying to compile CPU package, and I finally figured out that this is the line that causes the following error in my compilation:

```
In file included from /home/mzalewski/.pyenv/versions/miniforge3/conda-bld/cunumeric_1657756345722/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/include/cuda/std/cstddef:32,
                 from /home/mzalewski/.pyenv/versions/miniforge3/conda-bld/cunumeric_1657756345722/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/include/cuda/std/type_traits:17,
                 from /home/mzalewski/.pyenv/versions/miniforge3/conda-bld/cunumeric_1657756345722/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/include/curand_kernel.h:72,
                 from cunumeric/random/randutil/randutil_curand.h:95,
                 from cunumeric/random/randutil/generator.h:22,
                 from cunumeric/random/randutil/generator_host_straightforward.cc:17:
/home/mzalewski/.pyenv/versions/miniforge3/conda-bld/cunumeric_1657756345722/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/include/cuda/std/detail/libcxx/include/cstddef:58:9: error: 'ptrdiff_t' has not been declared in '::'
   58 | using ::ptrdiff_t;
      |         ^~~~~~~~~
compilation terminated due to -Wfatal-errors.
```